packages/babel-parser/src/plugins/flow.js

```diff
@@ -96,7 +96,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     }
 
     finishToken(type: TokenType, val: any): void {
-      if (!(type === tt.string || type === tt.semi)) {
+      if (![tt.string, tt.semi, tt.interpreterDirective].includes(type)) {
```

Comment:
Since this code will be run a really high number of times, maybe it's better to avoid the array allocation and the `includes` call?

```suggestion
      if (type !== tt.string && type !==  tt.semi && type !== tt.interpreterDirective) {
```